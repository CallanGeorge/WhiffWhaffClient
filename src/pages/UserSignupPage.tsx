import css from "./popup.module.css";
import { useRef, useState } from "react";
import axios from "axios";

export const UserSignupPage = () => {
  const nameRef = useRef<HTMLInputElement>(null);
  const passwordRef = useRef<HTMLInputElement>(null);
  const confirmPasswordRef = useRef<HTMLInputElement>(null);

  const [signUp, setSignup] = useState(false);
  const [error, setError] = useState("");
  const [pending, setPending] = useState(false);

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();

    if (!signUp) {
      axios
        .post("http://localhost:8080/api/v1/users", {
          auth: {
            username: nameRef.current!.value,
            password: passwordRef.current!.value,
          },
        })
        .then(() => setPending(false))
        .catch((err) => setError("Error loggin in"));
    }

    if (signUp) {
      !confirmPasswordRef.current!.value || !passwordRef.current!.value
        ? setError("Input password ")
        : confirmPasswordRef.current?.value !== passwordRef.current?.value
        ? setError("Passwords do not match")
        : (() => setPending(true)) &&
          (() => setError("")) &&
          axios
            .post(
              "http://localhost:8080/api/v1/users",

              {
                username: nameRef.current!.value,
                password: passwordRef.current!.value,
              }
            )
            .then(() => setPending(false))
            .catch((err) => setError("Username already in use."));
    }
  };

  // refactor this into smaller components

  return (
    <main className={css.container}>
      <h1 role="heading">Welcome to whiff whaff</h1>
      {signUp === false ? (
        <form className={css.form} onSubmit={handleSubmit}>
          <h2>Sign In! or</h2>
          <button type="button" onClick={() => setSignup(true)}>
            Sign Up
          </button>

          <label>Name</label>
          <input type="text" name="name" ref={nameRef} />
          <label>Password</label>
          <input type="password" name="password" ref={passwordRef} />
          <button type="submit">Submit</button>
        </form>
      ) : (
        <form className={css.form} onSubmit={handleSubmit}>
          <h2>Sign Up! or</h2>
          <button type="button" onClick={() => setSignup(false)}>
            Sign In
          </button>

          <label>Name</label>
          <input type="text" name="name" ref={nameRef} />
          <label>Password</label>
          <input type="password" name="password" ref={passwordRef} />
          <label>Confirm Password</label>
          <input type="password" name="password" ref={confirmPasswordRef} />
          <button type="submit">
            {pending ? <span>Pending</span> : <span>Submit</span>}
          </button>
          <p>{error}</p>
        </form>
      )}
    </main>
  );
};
