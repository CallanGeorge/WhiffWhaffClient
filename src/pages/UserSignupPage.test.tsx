import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { UserSignupPage } from "./UserSignupPage";

describe("UserSignupPage.t", () => {
  describe("Layout", () => {
    it("has header", async () => {
      const { container } = render(<UserSignupPage />);
      await screen.findByRole("heading", { name: "UserSignupPage" });
      expect(container).toMatchSnapshot();
    });
  });
});
