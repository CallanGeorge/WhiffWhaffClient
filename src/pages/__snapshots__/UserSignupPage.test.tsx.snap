// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSignupPage.t Layout has header 1`] = `
<div>
  <h1
    role="heading"
  >
    UserSignupPage
  </h1>
</div>
`;
